src/clientSideScene/sceneEntities.ts

```diff
@@ -1584,6 +1667,448 @@ export class SceneEntities {
+        const sketchInit = _node.node?.declaration.init
+
+        let modded = structuredClone(_ast)
+        if (sketchInit.type === 'PipeExpression') {
+          // Calculate end angle based on final mouse position
+          const endAngle = Math.atan2(
+            mousePoint.y - center[1],
+            mousePoint.x - center[0]
+          )
+
+          // Calculate the final 'to' point using the existing radius and the final end angle
+          const finalTo: [number, number] = [
+            center[0] + radius * Math.cos(endAngle),
+            center[1] + radius * Math.sin(endAngle),
+          ]
+
+          const moddedResult = changeSketchArguments(
+            modded,
+            kclManager.variables,
+            {
+              type: 'path',
+              pathToNode: mod.pathToNode,
+            },
+            {
+              type: 'arc-segment',
+              from: lastSeg.to,
+              to: finalTo,
+              center: center,
+              radius: radius,
+              ccw: true,
+            }
+          )
+          if (err(moddedResult)) return
+          modded = moddedResult.modifiedAst
+
+          const newCode = recast(modded)
+          if (err(newCode)) return
+          const pResult = parse(newCode)
+          if (trap(pResult) || !resultIsOk(pResult))
+            return Promise.reject(pResult)
+          _ast = pResult.program
+
+          // Update the primary AST and unequip the arc tool
+          await kclManager.executeAstMock(_ast)
+          sceneInfra.modelingSend({ type: 'Finish arc' })
+          await codeManager.updateEditorWithAstAndWriteToFile(_ast)
+        }
+      },
+    })
+    return {
+      updatedEntryNodePath: sketchEntryNodePath,
+      updatedSketchNodePaths: sketchNodePaths,
+      expressionIndexToDelete: -1, // No need to delete any expression
+    }
+  }
+  setupDraftArcThreePoint = async (
+    sketchEntryNodePath: PathToNode,
+    sketchNodePaths: PathToNode[],
+    planeNodePath: PathToNode,
+    forward: [number, number, number],
+    up: [number, number, number],
+    sketchOrigin: [number, number, number],
+    p2: [x: number, y: number]
+  ): Promise<SketchDetailsUpdate | Error> => {
+    let _ast = structuredClone(kclManager.ast)
+
+    const _node1 = getNodeFromPath<VariableDeclaration>(
+      _ast,
+      sketchEntryNodePath || [],
+      'VariableDeclaration'
+    )
+    if (trap(_node1)) return Promise.reject(_node1)
+    const variableDeclarationName = _node1.node?.declaration.id?.name || ''
+
+    const sg = sketchFromKclValue(
+      kclManager.variables[variableDeclarationName],
+      variableDeclarationName
+    )
+    if (err(sg)) return Promise.reject(sg)
+    const lastSeg = sg?.paths?.slice(-1)[0] || sg.start
+
+    // Calculate a default center point and radius based on the last segment's endpoint
+    const p1: [number, number] = [lastSeg.to[0], lastSeg.to[1]]
+    const p3: [number, number] = [p2[0] + 0.1, p2[1] + 0.1]
```

Comment:
Could we make this adjustable? (0.1) Or not worth it you think?

Reply:
This is just an arbitary point as a starting point for animating the draft arc while waiting for the user ta make the final click, not sure what make it adjustable achieves, where and who would be doing the adjusting?